systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -271,8 +272,24 @@ public void deleteResources(ExtensionContext testContext) throws Exception {
         if (!STORED_RESOURCES.containsKey(testContext.getDisplayName()) || STORED_RESOURCES.get(testContext.getDisplayName()).isEmpty()) {
             LOGGER.info("In context {} is everything deleted.", testContext.getDisplayName());
         }
-        while (STORED_RESOURCES.containsKey(testContext.getDisplayName()) && !STORED_RESOURCES.get(testContext.getDisplayName()).isEmpty()) {
-            STORED_RESOURCES.get(testContext.getDisplayName()).pop().getThrowableRunner().run();
+
+
+        // if stack is created for specific test suite or test case
+        AtomicInteger numberOfResources = STORED_RESOURCES.get(testContext.getDisplayName()) != null ?
+            new AtomicInteger(STORED_RESOURCES.get(testContext.getDisplayName()).size()) :
+            // stack has no elements
+            new AtomicInteger(0);
+        while (STORED_RESOURCES.containsKey(testContext.getDisplayName()) && numberOfResources.get() > 0) {
+            STORED_RESOURCES.get(testContext.getDisplayName()).parallelStream().parallel().forEach(
+                resourceItem -> {
+                    try {
+                        resourceItem.getThrowableRunner().run();
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
Wouldn't be better to log something into the console? Maybe debug level could be enough

Reply:
This automatically invokes the delete() method:
```
2021-11-24 11:56:32 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of CustomResourceDefinition kafkamirrormaker2s.kafka.strimzi.io in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of CustomResourceDefinition kafkarebalances.kafka.strimzi.io in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ConfigMap strimzi-cluster-operator in namespace infra-namespace
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-broker-delegation in namespace (not set)
2021-11-24 11:56:42 [main] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-client-delegation in namespace (not set)
2021-11-24 11:56:42 [main] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-broker-delegation in namespace (not set)
2021-11-24 11:56:42 [ForkJoinPool.commonPool-worker-5] INFO  [ResourceManager:196] Delete of RoleBinding strimzi-cluster-operator in namespace infra-namespace
2021-11-24 11:56:42 [main] INFO  [ResourceManager:196] Delete of ClusterRoleBinding strimzi-cluster-operator-kafka-client-delegation in namespace (not set)
```